react/components/Billing/components/BillingForm/index.js

```diff
@@ -0,0 +1,302 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { propType } from 'graphql-anywhere';
+import { injectStripe } from 'react-stripe-elements';
+import { compose, graphql } from 'react-apollo';
+
+import mapErrors from 'react/util/mapErrors';
+
+import subscribeToPremiumMutation from 'react/components/Billing/components/BillingForm/mutations/subscribeToPremium';
+import cancelPremiumMutation from 'react/components/Billing/components/BillingForm/mutations/cancelPremium';
+import addCreditCardMutation from 'react/components/Billing/components/BillingForm/mutations/addCreditCard';
+
+import billingFormFragment from 'react/components/Billing/components/BillingForm/fragments/billingForm';
+
+import Box from 'react/components/UI/Box';
+import Alert from 'react/components/UI/Alert';
+import ErrorAlert from 'react/components/UI/ErrorAlert';
+import LoadingIndicator from 'react/components/UI/LoadingIndicator';
+import GenericButton from 'react/components/UI/GenericButton';
+import { LabelledInput, Label } from 'react/components/UI/Inputs';
+import PlanSelection from 'react/components/Billing/components/PlanSelection';
+import CouponCode from 'react/components/Billing/components/CouponCode';
+import MyCreditCard from 'react/components/Billing/components/MyCreditCard';
+import PlanChanges from 'react/components/Billing/components/PlanChanges';
+import CancellationNotice from 'react/components/Billing/components/CancellationNotice';
+import CancelPremium from 'react/components/Billing/components/CancelPremium';
+import StatusOverlay from 'react/components/Billing/components/StatusOverlay';
+
+const OPERATIONS = {
+  ADD_NEW_CREDIT_CARD: 'ADD_NEW_CREDIT_CARD',
+  CHANGE_DEFAULT_CREDIT_CARD: 'CHANGE_DEFAULT_CREDIT_CARD',
+  CHANGE_PLAN_ID: 'CHANGE_PLAN_ID',
+  REENABLE_PREMIUM: 'REENABLE_PREMIUM',
+  CANCEL_PREMIUM: 'CANCEL_PREMIUM',
+};
```

Comment:
Working on this now; trying to clean up the state. I hate that I'm realizing I basically need a state machine at this point.